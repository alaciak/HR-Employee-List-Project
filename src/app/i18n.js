import i18n from 'i18next';
import Backend from 'i18next-xhr-backend';
import { reactI18nextModule } from 'react-i18next';

i18n
  .use(Backend)
  .use(reactI18nextModule)
  .init({
    fallbackLng: 'en',
    ns: ['translations'],
    defaultNS: 'translations',
    debug: true,
    react: {
      wait: true
    }
  });


export default i18n;
