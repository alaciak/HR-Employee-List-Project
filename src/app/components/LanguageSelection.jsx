import React from 'react';
import { translate} from 'react-i18next';
import '../i18n';

export class LanguageSelection extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedLanguage: 'en'
    }
  }

  handleOnClick = (i18n, language) => {
    i18n.changeLanguage(language);
    this.setState({
      selectedLanguage: language
    });
  }

  render() {
    const selectedEnglish = (this.state.selectedLanguage === 'en')? '#e5a357' : '#fcfaf9';
    const selectedPolish = (this.state.selectedLanguage === 'pl')? '#e5a357' : '#fcfaf9';
    return (
      <div className='navigation-language col-2'>
        <button type='button' style={{ color: selectedEnglish }} onClick={ () => this.handleOnClick(this.props.i18n, 'en') }>ENG</button>
        <button type='button' style={{ color: selectedPolish }} onClick={ () => this.handleOnClick(this.props.i18n, 'pl') }>PL</button>
      </div>
    );
  }
}

export default (translate('translations')(LanguageSelection));
