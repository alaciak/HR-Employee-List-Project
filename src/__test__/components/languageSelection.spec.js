import React from 'react';
import { LanguageSelection } from '../../app/components/LanguageSelection.jsx';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import i18n from '../i18nForTests';


describe('LanguageSelection', () => {

  it('should render LanguageSelection component', () => {

    const wrapper = shallow(<LanguageSelection />);

    expect(toJson(wrapper)).toMatchSnapshot();

  });

});
