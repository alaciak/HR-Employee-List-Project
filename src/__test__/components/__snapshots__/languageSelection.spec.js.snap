// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelection should render LanguageSelection component 1`] = `
<div
  className="navigation-language col-2"
>
  <button
    onClick={[Function]}
    style={
      Object {
        "color": "#e5a357",
      }
    }
    type="button"
  >
    ENG
  </button>
  <button
    onClick={[Function]}
    style={
      Object {
        "color": "#fcfaf9",
      }
    }
    type="button"
  >
    PL
  </button>
</div>
`;
